subprojects/core/src/main/java/org/gradle/process/internal/streams/ExecOutputHandleRunner.java

```diff
@@ -62,7 +63,7 @@ public void run() {
     }
 
     public void run(Executor executor) {
-        executor.execute(this);
+        executor.execute(new BuildOperationIdentifierPreservingRunnable(this));
```

Comment:
I decided to use this method in `StreamsForwarder#start()` as it avoided code duplication and the method was already here. What would you suggest?